scripts/create_project_table.py

```diff
@@ -0,0 +1,131 @@
+"""
+Process README files to extract metadata, generate a table of content, 
+and update a central README file with aggregated data.
+
+- Extract metadata from the top section of the README file for each project (e.g., between "---" delimiters).
+- Generate table rows summarizing the metadata
+- Insert the generated rows into the project table in the main README file
+
+Classes:
+- ReadmeMetadataProcessor: Extracts and processes metadata from individual README files.
+"""
+
+import os
+import yaml
+from pathlib import Path
+from typing import List
+
+ROOT_PATH = Path(__file__).parent.parent
+README_PATH = ROOT_PATH / "README.md"
+
+
+class ReadmeMetadataProcessor:
+    def __init__(self, readme_file: Path):
+        self.readme_file = readme_file
+        self.metadata = self.extract_metadata()
+
+    def extract_metadata(self) -> dict:
+        metadata = {}
+        lines = self.readme_file.read_text(encoding="utf-8").splitlines()
+
+        # Check for metadata block at the top of the README
+        if lines[0].strip() == "---":
+            metadata_lines = []
+            for line in lines[1:]:
+                if line.strip() == "---":
+                    break
+                metadata_lines.append(line)
+
+            metadata = yaml.safe_load("\n".join(metadata_lines))
+
+        return metadata
+
+    def get_title(self) -> str:
+        return self.metadata.get("title", None)
+
+    def get_description(self) -> str:
+        return self.metadata.get("description", "No description available.")
```

Comment:
1. Return `""` instead of `None` and `"No description available."
2. These function's aren't needed - `dict.get(key, "")` is a trivial call that doesn't need wrapping